cmd/contour/shutdownmanager.go

```diff
@@ -277,11 +288,12 @@ func registerShutdownManager(cmd *kingpin.CmdClause, log logrus.FieldLogger) (*k
 }
 
 // registerShutdown registers the envoy shutdown sub-command and flags
-func registerShutdown(cmd *kingpin.CmdClause, log logrus.FieldLogger) (*kingpin.CmdClause, *shutdownContext) {
-	ctx := newShutdownContext()
+func registerShutdown(cmd *kingpin.CmdClause, defaultAdminPort int, log logrus.FieldLogger) (*kingpin.CmdClause, *shutdownContext) {
+	ctx := newShutdownContext(defaultAdminPort)
 	ctx.FieldLogger = log.WithField("context", "shutdown")
 
 	shutdown := cmd.Command("shutdown", "Initiate an shutdown sequence which configures Envoy to begin draining connections.")
+	shutdown.Flag("admin-port", "Envoy admin interface port.").IntVar(&ctx.adminPort)
```

Comment:
```suggestion
	shutdown.Flag("admin-port", "Envoy admin interface port.").Default("9001").IntVar(&ctx.adminPort)
```

Should be able to default the value here which removes the need for the check on https://github.com/projectcontour/contour/pull/3501/files#diff-ab19309345ae701385f9f0a7315157a1a76ad909f3dacef2e8ad9fb973a4232fR90